<?xml version="1.0"?>
<launch>
    <arg name="agentNum" default="1" />
    <arg name="sim" default="false" />
    <arg name="use_rosbag" default="true" />
    <arg name="central_yaml" default="$(find angle_aware_control)/config/central.yaml" />

    <rosparam file="$(arg central_yaml)" command="load" ns="" />

    <node pkg="bebop_hatanaka_base" type="uh_from_joy.py" name="uh_from_joy" output="log" />

    <include file="$(find bebop_hatanaka_base)/launch/central_base.launch">
        <arg name="agentNum" value="$(arg agentNum)" />
        <arg name="sim" value="$(arg sim)" />
        <arg name="rviz_file" value="$(find angle_aware_control)/config/sim_$(arg sim).rviz" />
        <arg name="use_rosbag" value="$(arg use_rosbag)" />
        <arg name="rosbag_output" default="$(find angle_aware_control)/rosbag/rosbag.bag" />
    </include>

    <node pkg="angle_aware_control" type="central.py" name="central" output="screen" />
    <node pkg="coverage_util" type="show_pointcloud2d.py" name="show_pointcloud2d" output="screen">
        <param name="input_field_topic" value="psi" />
        <param name="field_param_name" value="angle_aware/psi" />
        <param name="rviz_z_max" value="0.5" />
    </node>

    <rosparam file="$(find angle_aware_control)/config/AR.yaml" command="load" ns="" />

</launch>